pkgs/development/libraries/ncurses/default.nix

```diff
@@ -27,8 +27,6 @@ stdenv.mkDerivation rec {
     sha256 = "15r2456g0mlq2q7gh2z52vl6zv6y0z8sdchrs80kg4idqd8sm8fd";
   };
 
-  patches = lib.optional (!stdenv.cc.isClang) ./clang.patch;
```

Comment:
Isn't this strange anyway? If we're not on clang, apply a clang patch?

Reply:
Yeah this is what confused me as well, but I've traced back this line through every change to it and its introduction and this seems to be intentional. I also checked that compilation with gcc and this patch actually fails.

Reply:
Of course I meant: applying this with clang actually breaks the build.